### ISSUE TITLE
`valid-jsdoc` rule: Allow "event:" within `@param` types

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#11036]
**Tell us about your environment**

* **ESLint Version:** 5.8.0
* **Node Version:** 9.2.1
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 5,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "valid-jsdoc": 2
    },
    "env": {}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I used "event:" as part of a string comprising an event name reference within `@param`.

See https://eslint.org/demo/#eyJ0ZXh0IjoidmFyIGV4cG9ydHMgPSB7XG4gICAgLyoqXG4gICAgICogUmVwb3J0IHRoZSB0aHJvd2luZyBvZiBhIHNub3diYWxsLlxuICAgICAqXG4gICAgICogQG1ldGhvZFxuICAgICAqIEBwYXJhbSB7bW9kdWxlOmh1cmxlcn5zbm93YmFsbH0gZSAtIEEgc25vd2JhbGwgZXZlbnQuXG4gICAgICogQGxpc3RlbnMgbW9kdWxlOmh1cmxlcn5zbm93YmFsbFxuICAgICAqIEByZXR1cm5zIHt1bmRlZmluZWR9XG4gICAgICovXG4gICAgcmVwb3J0VGhyb3dhZ2U6IGZ1bmN0aW9uIChlKSB7XG4gICAgICAgIHRoaXMubG9nKCdzbm93YmFsbCB0aHJvd246IHZlbG9jaXR5ICcgKyBlLnZlbG9jaXR5KTtcbiAgICB9XG59O1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7InZhbGlkLWpzZG9jIjoyfSwiZW52Ijp7fX19

```js
var exports = {
    /**
     * Report the throwing of a snowball.
     *
     * @method
     * @param {module:hurler~event:snowball} e - A snowball event.
     * @listens module:hurler~event:snowball
     * @returns {undefined}
     */
    reportThrowage: function (e) {
        this.log('snowball thrown: velocity ' + e.velocity);
    }
};
```

No special command used to run ESLint here; just the demo (though getting it from `eslint .` locally as well).

**What did you expect to happen?**

The [JSDocs (for @listens)](http://usejsdoc.org/tags-listens.html) states:

> The following example shows how to document an event named module:hurler~event:snowball, as well as a method named module:playground/monitor.reportThrowage that listens for the event...

So "event:" should be permitted within references to events (including `@param`) without complaint, at least at preceding a property name at the beginning of [namepaths](http://usejsdoc.org/about-namepaths.html#namepaths-in-jsdoc-3) (`#`, `.`, or `~` for JSDoc 3, and apparently `-`, if supporting JSDoc 2), so, e.g., `MyClass#event:foo` or `MyClass~event:foo`.

The code above was even taken from http://usejsdoc.org/tags-listens.html (with `@returns` added to avoid the error about it missing `@returns`). This code uses `event:` within a `@param`, and if this is removed, ESLint ceases to report an error (it does not care about the `@listens` use of this though).

Even if one could work around this by avoiding "event:" within event names, it is not only explicitly permitted by JSDoc but is helpfully descriptive in distinguishing such events from other object references.

Moreover per http://usejsdoc.org/tags-event.html , it appears it may be even required in some locations (and is explicitly permitted in the signature):

>  **Syntax**
>
> @event &lt;className>#[event:]&lt;eventName> 
> ...
> JSDoc automatically prepends the namespace event: to each event's name. In general, you must include this namespace when you link to the event in another doclet. (The @fires tag is a notable exception; it allows you to omit the namespace.)

**What actually happened? Please include the actual, raw output from ESLint.**

> 2:5 - JSDoc syntax error. (valid-jsdoc)

**Are you willing to submit a pull request to fix this bug?**

No, not for the foreseeable future.

**Comment by nzakas** [comment]
Thanks for the issue. It looks like the problem here is that Doctrine, the JSDoc parser we use, just doesn't support this syntax. The only way to fix this would be to make a fix to Doctrine itself. Can you [open an issue](https://github.com/eslint/doctrine/issues/new) there instead?
